osbenchmark/metrics.py

```diff
@@ -1428,7 +1428,7 @@ def as_dict(self):
             }
         }
         if self.results:
-            d["results"] = self.results.as_dict()
+            d["results"] = self.results if isinstance(self.results, dict) else self.results.as_dict()
```

Comment:
Wondering your guys thoughts on this. I figured since it needs to be a dict anyway that we can write it to accept results that are already dicts

Reply:
Will there be cases where self.results is already a dict? I don't think it hurts to reinforce it

Reply:
Makes sense. Reverted this change